Plutarch/Builtin.hs

```diff
@@ -275,3 +298,13 @@ pinnerData = punsafeCoerce
 
 pouterData :: Term s (PAsData (PInner a b)) -> Term s (PAsData a)
 pouterData = punsafeCoerce
+
+pconstrBuiltin :: Term s (PInteger :--> PBuiltinList PData :--> PAsData (PBuiltinPair PInteger (PBuiltinList PData)))
+pconstrBuiltin = punsafeBuiltin $ PLC.ConstrData
+
+{- | Create a Plutarch-level 'PAsData' constant, from a Haskell value.
+Example:
+> pconstantData @PInteger 42
+-}
+pconstantData :: forall p h s. (ToData h, PLifted p ~ h, PConstanted h ~ p) => h -> Term s (PAsData p)
```

Comment:
Apparently this can't build bytestring constants (but it can build newtypes to bytestrings such as `PCurrencySymbol`). This is because `ByteString` doesn't have a `ToData` instance - only `BuiltinByteString` does.....